node/node_newblock.go

```diff
@@ -82,8 +82,16 @@ func (node *Node) proposeNewBlock() (*types.Block, error) {
 
 	// Prepare transactions including staking transactions
 	selectedTxs, selectedStakingTxs := node.getTransactionsForNewBlock(coinbase)
-
-	if err := node.Worker.CommitTransactions(selectedTxs, selectedStakingTxs, coinbase); err != nil {
+	if err := node.Worker.CommitTransactions(
+		selectedTxs, selectedStakingTxs, coinbase,
+		func(payload []types.RPCTransactionError) {
+			node.errorSink.Lock()
```

Comment:
do you need to unlock?

Reply:
ah, good catch, copy paste from master commit was cut off , fixed